satpy/readers/sar_c_safe.py

```diff
@@ -704,3 +689,85 @@ def start_time(self):
     def end_time(self):
         """Get the end time."""
         return self._end_time
+
+
+class SAFESARReader(GenericYAMLReader):
+    """A reader for SAFE SAR-C data for Sentinel 1 satellites."""
+
+    def __init__(self, config, filter_parameters=None):
+        """Set up the SAR reader."""
+        super().__init__(config)
+        self.filter_parameters = filter_parameters
+        self.files_by_type = defaultdict(list)
+        self.storage_items = []
+
+    @property
+    def start_time(self):
+        """Get the start time."""
+        return self.storage_items.values()[0].filename_info["start_time"].replace(tzinfo=tz.utc)
+
+    @property
+    def end_time(self):
+        """Get the end time."""
+        return self.storage_items.values()[0].filename_info["end_time"].replace(tzinfo=tz.utc)
```

Comment:
These look a bit costly (maybe, see my comment on storage item collection), should these be cached properties?

Reply:
I'm not sure about this, as in theory the reader object is not immutable... a cached property would make the assumption that the files/storage items never changes.
I know that in practice, this never happens, but this may still be confusing and bug-prone

Reply:
no, it's the same compared from the reader level (not the filehandler). In theory, nothing prevents a user to call `create_filehandlers` (now `create_storage_items`) multiple times on the same reader...